File: LearnObjectiveC/Objective-C-Ex1/Objective-C-Ex1/HomeViewController.m
//
//  HomeViewController.m
//  Objective-C-Ex1
//
//  Created by Mylo Ho on 9/12/16.
//  Copyright © 2016 Ho Van Su. All rights reserved.
//

#import "HomeViewController.h"

@interface HomeViewController () {
    
bool isStop;
    
}

@end

@implementation HomeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    isStop = false;
    NSTimer *timer = [NSTimer timerWithTimeInterval:1 target:self selector:@selector(setTime) userInfo:nil repeats:YES];
    [[NSRunLoop mainRunLoop] addTimer:timer forMode:NSRunLoopCommonModes];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    UITouch *touch = [[event allTouches] anyObject];
    if (touch.view == self.view) {
        isStop = !isStop;
    }
}

-(void) setTime {
    if (!isStop) {
        self.helloWorldLabel.text = [NSString stringWithFormat:@"%@", [NSDate date]];
    }
}

@end

File: iOSDay4/iOSEx4Timer/iOSEx4Timer/HomeViewController.swift
//
//  HomeViewController.swift
//  iOSEx4Timer
//
//  Created by Mylo Ho on 6/16/16.
//  Copyright © 2016 Ho Van Su. All rights reserved.
//

import UIKit

class HomeViewController: UIViewController {

    @IBOutlet weak var timeSystem: UILabel!
    
    var timer = NSTimer()
    var isRun = true
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.timer = NSTimer.scheduledTimerWithTimeInterval(0.5, target: self, selector: #selector(self.setTime), userInfo: nil, repeats: true)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
        if isRun {
            timer.invalidate()
            isRun = false
        } else {
            timer = NSTimer.scheduledTimerWithTimeInterval(0.5, target: self, selector: #selector(self.setTime), userInfo: nil, repeats: true)
            isRun = true
        }
    }
    
    func convertDateToString(date: NSDate) -> String {
        let format = NSDateFormatter()
        format.dateFormat = "HH:mm:ss"
        
        let dateString = format.stringFromDate(date)
        
        return dateString
    }
    
    func setTime() {
        self.timeSystem.text = "\(self.convertDateToString(NSDate()))"
    }


}
